File: programers_Lv2/[3차] n진수 게임.py
# https://school.programmers.co.kr/learn/courses/30/lessons/17687?language=python3
def solution(n, t, m, p):
    num_list = [
        "0",
        "1",
        "2",
        "3",
        "4",
        "5",
        "6",
        "7",
        "8",
        "9",
        "A",
        "B",
        "C",
        "D",
        "E",
        "F",
    ]

    new_num_list = num_list[:n]

    result_str = "0"
    end = p + (t - 1) * m
    for i in range(1, end):
        num = ""
        quotient = i

        while quotient > 0:
            remainder = quotient % n
            num = new_num_list[remainder] + num
            quotient = quotient // n

        result_str += num

    result_str = result_str[p - 1 :: m][:t]
    return result_str


print(solution(2, 4, 2, 1))  # "0111"

File: programers_Lv2/[3차] n진수 게임.js
// https://school.programmers.co.kr/learn/courses/30/lessons/17687?language=python3
function solution(n, t, m, p) {
  const numArr = [
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
    "A",
    "B",
    "C",
    "D",
    "E",
    "F",
  ];

  const newNumArr = numArr.slice(0, n);

  let resultStr = "0";
  let end = p + (t - 1) * m;
  for (let i = 1; i < end; i++) {
    let num = "";
    let quotient = i;

    while (quotient > 0) {
      let remainder = quotient % n;
      num = newNumArr[remainder] + num;
      quotient = Math.floor(quotient / n);
    }

    resultStr += num;
  }

  const result = Array.from(resultStr.slice(p - 1, -1))
    .filter((_, index) => index % m === 0)
    .join("");

  return result.slice(0, t);
}

// console.log(solution(2, 4, 2, 1)); // "0111"
console.log(solution(16, 16, 2, 2)); // "13579BDF01234567"
